src/vivarium/framework/time.py

```diff
@@ -122,20 +139,23 @@ def step_backward(self) -> None:
     def step_forward(self, index: pd.Index) -> None:
         """Advances the clock by the current step size, and updates aligned simulant clocks."""
         self._clock_time += self.step_size
-        pop_to_update = self.get_active_population(index, self.time)
-        if not pop_to_update.empty:
-            pop_to_update["step_size"] = self.step_size_pipeline(pop_to_update.index)
-            pop_to_update["next_event_time"] = self.time + pop_to_update["step_size"]
-            self.population_view.update(pop_to_update)
-        self._clock_step_size = self.simulant_next_event_times(index).min() - self.time
-
-    def get_active_population(self, index: pd.Index, time: Time):
+        if self.individual_clocks:
+            update_index = self.get_active_population(index, self.time)
+            pop_to_update = self.individual_clocks.get(update_index)
```

Comment:
Nit: `pop_to_update` as a variable name seems like an artifact of when `self.individualized_clocks` was called `self.population_view`.

Reply:
renamed to get_active_simulants and clocks_to_update